udata_hydra/analysis/csv.py

```diff
@@ -179,9 +182,22 @@ async def csv_to_db(file_path: str, inspection: dict, table_name: str, debug_ins
     :table_name: used to create tables
     :debug_insert: insert record one by one instead of using postgresql COPY
     """
+    def generate_records(file_path, inspection, columns):
+        # because we need the iterator twice, not possible to
+        # handle parquet and db through the same iteration
+        def _records():
```

Comment:
It wasn't working otherwise if I remember it correctly 😅 we're going through the iterator twice so we recreate it from scratch when the function is called

Reply:
Actually you were right 🙏